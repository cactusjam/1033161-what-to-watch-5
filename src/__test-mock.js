import {NameSpace} from "./store/reducers/root-reducer.js";

export const TEST_MOCK_MOVIE = {
  background: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  backgroundColor: `#73B39A`,
  cover: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
  director: `Nicolas Winding Refn`,
  duration: 92,
  genre: `Action`,
  id: 1,
  isFavorite: true,
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  promo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: {
    score: 3.6,
    countOfVotes: 109661
  },
  releaseYear: 2008,
  starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  title: `Bronson`,
  video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`
};

export const TEST_MOCK_COMMENT = {
  author: `Christina`,
  date: `2020-10-09T13:38:44.769Z`,
  id: 1,
  rating: 3.3,
  reviewText: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quali.`
};

export const TEST_MOCK_USER = {
  id: 1,
  email: `qqq@gmail.com`,
  userAvatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
  authorizationStatus: `AUTH`,
  password: `666`,
  errorEmail: ``,
  errorPassword: ``
};

export const TEST_MOCKS = {
  id: 1,
  noop: () => {},
  movie: TEST_MOCK_MOVIE,
  movies: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
  genresFilter: [`All genres`, `Action`, `Adventure`, `Comedy`, `Crime`, `Drama`, `Fantasy`, `Thriller`],
  genreFilter: `All genres`,
  review: TEST_MOCK_COMMENT,
  userData: {
    id: 1,
    email: `qqq@gmail.com`,
    userAvatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
    authorizationStatus: `AUTH`
  },
  unauthorizedUserData: {
    id: 1,
    email: ``,
    userAvatar: ``,
    authorizationStatus: `NO_AUTH`
  },
  reviews: [TEST_MOCK_COMMENT, TEST_MOCK_COMMENT, TEST_MOCK_COMMENT],
};

export const TEST_MOCK_STORE = {
  [NameSpace.USER]: {
    id: 1,
    email: `qqq@gmail.com`,
    authorizationStatus: `AUTH`,
    userAvatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
  },
  [NameSpace.GENRES]: {
    activeFilterGenre: `Adventure`
  },
  [NameSpace.DATA]: {
    movies: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
    reviews: [TEST_MOCK_COMMENT, TEST_MOCK_COMMENT, TEST_MOCK_COMMENT],
    currentMovie: TEST_MOCK_MOVIE,
    promoMovie: TEST_MOCK_MOVIE,
    favorites: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
    defaultFilmsCount: 8,
    isDataSending: false,
    isDataSendError: false
  },
};
