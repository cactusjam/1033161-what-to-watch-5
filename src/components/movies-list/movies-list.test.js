import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./movies-list";
import {TEST_MOCK_STORE, TEST_MOCKS} from "../../__test-mock.js";
import {MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore(TEST_MOCK_STORE);

describe(`MoviesList`, () => {
  it(`Should MoviesList render correctly`, () => {
    const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <MoviesList
              movies={TEST_MOCKS.movies}
            />,
          </MemoryRouter>
        </Provider>
        , {
          createNodeMock: () => {
            return {};
          }
        }
    )
  .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
