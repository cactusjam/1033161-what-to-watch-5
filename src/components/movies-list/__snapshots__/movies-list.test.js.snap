// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList Should MoviesList render correctly 1`] = `
Array [
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      id={1}
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            autoPlay={false}
            height="175"
            muted={true}
            poster=""
            src=""
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <span
            className="small-movie-card__link"
          >
            
          </span>
        </h3>
      </a>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      id={1}
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            autoPlay={false}
            height="175"
            muted={true}
            poster=""
            src=""
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <span
            className="small-movie-card__link"
          >
            
          </span>
        </h3>
      </a>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      id={1}
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            autoPlay={false}
            height="175"
            muted={true}
            poster=""
            src=""
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <span
            className="small-movie-card__link"
          >
            
          </span>
        </h3>
      </a>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      id={1}
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            autoPlay={false}
            height="175"
            muted={true}
            poster=""
            src=""
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <span
            className="small-movie-card__link"
          >
            
          </span>
        </h3>
      </a>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      id={1}
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        <div
          className="small-movie-card__image"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <video
            autoPlay={false}
            height="175"
            muted={true}
            poster=""
            src=""
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <span
            className="small-movie-card__link"
          >
            
          </span>
        </h3>
      </a>
    </article>
  </div>,
  ",",
]
`;
